### PR TITLE
newsboat: add passthru.updateScript

### PR DESCRIPTION
**Pull request by nicknovitski**
###### Description of changes

Very simple, and also tested by running `nix-shell maintainers/scripts/update.nix --argstr package newsboat`.  I didn't include the actual update to the latest version since @dotlambda has the PR for that, #223326

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
Why do we need an update script if `nix-update` works just fine?

**Comment by sikmir** [comment]
I guess nix-update doesn't use `passthru.updateScript`, but @r-ryantm does.

**Comment by trofi** [comment]
@ofborg eval

**Comment by AndersonTorres** [comment]
@ofborg eval

File: pkgs/applications/networking/feedreaders/newsboat/default.nix [hunk 1]
```diff
@@ -55,9 +55,14 @@ rustPlatform.buildRustPackage rec {
...
   '';
 
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
   meta = with lib; {
     homepage    = "https://newsboat.org/";
     description = "A fork of Newsbeuter, an RSS/Atom feed reader for the text console";
+    changelog   = "https://github.com/newsboat/newsboat/blob/r${version}/CHANGELOG.md";
```
Comment (trofi) [inline pkgs/applications/networking/feedreaders/newsboat/default.nix:65]: Can you rebase? `ofborg` says `changelog` is already present there:

```
       error: attribute 'changelog' already defined at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/applications/networking/feedreaders/newsboat/default.nix:64:5

       at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-2/pkgs/applications/networking/feedreaders/newsboat/default.nix:66:5:

           65|     description = "A fork of Newsbeuter, an RSS/Atom feed reader for the text console";
           66|     changelog   = "https://github.com/newsboat/newsboat/blob/r${version}/CHANGELOG.md";
             |     ^
```

**Comment by AndersonTorres** [comment]
Is there any unwritten directive about creating those updaters?

**Review by trofi — APPROVED** [review]

**Pull request merged**